packages/common/src/doctype.ts

```diff
@@ -70,6 +70,10 @@ export interface DocNext {
     metadata?: DocMetadata;
 }
 
+export interface LogEntry {
+  cid: CID
+  isVersion?: boolean
```

Comment:
Great improvement making the log array contain structured objects rather than just the CID!

I wonder if instead of an isVersion bool we should store something representing the type of Record (GenesisRecord, SignedRecord, AnchorRecord) this entry represents.  Then when filtering the log for entries that are also versions we filter for Genesis or Anchor Records.  Gives us more information in the log, and pushes the logic for deciding which entries are versions from where the log entries are built to where we are actually requesting the set of versions, which puts all the code for dealing with versions in one place.

Reply:
This makes sense. I'll make an update.